Built a basic UI and a new example maze.

ZVK opened the pull request:
While attempting to integrate multiple mazes, I was able to stop the metronome and allow for a new JSON file to be loaded. I'm not getting any errors, but the songs are interrupting each other a bit when first changed. You can check out a live demo at [DadaBots](http://dadabots.com/MazeMetal/)

ZVK committed:
Added Maze-I Demo

Created a new demo song (maze-I.mp3) and configured mazemetal.js to load
JSON for it.

ZVK committed:
Created basic UI for changing mazes

added bootstrap, jquery, custom CSS theme, and JSON load function for
launching a new SoundEngine.

ZVK committed:
added new musical section

Attempting to make the composition more interesting when looped.

Cortexelus commented:
Awesome! Interruption can be fixed with sound.stopAll()  (and/or stopAll needs to be fixed)

Also the browser puts a limit on the # of webkitAudioContexts that you can make; so SoundEngine should probably only be created once, and there should be a method for loading a new mp3.

I can fix these things when i get a moment

ZVK committed:
.stopAll error handler/clean code for "strict mode"

Corrected syntax for conform to "strict mode". Allowed for choice of
depreciated webkitAudioContext or the new AudioContext. Utilized
try/catch pattern for cases where audio has already stopped.

ZVK committed:
reverted to original code

removed loadMaze function to reflect upstream code.

ZVK committed:
added load method for new mp3s

ZVK committed:
function for reloading JSON and terminating jobs

This approach to changing songs makes the initial setup calls and
requests automatically. Then, All subsequent song requests (triggered by
changing the dropdown) cause the previous processes to be
terminated/aborted/stopped.

ZVK committed:
First commit for Auto Generating maze

generates ASCII maze for each new song.

ZVK committed:
Included Chris Wilson's Monkey Patch

This script allows us to use modern Web Audio API syntax in old
browsers.

ZVK committed:
Revert "Included Chris Wilson's Monkey Patch"

This reverts commit f95520b9f74d7b01eba8685bb6168c38d8bb17a0.
